src/runtime_src/core/edge/drm/zocl/zocl_ioctl.c

```diff
@@ -294,69 +258,24 @@ static int zocl_pcap_download(struct drm_zocl_dev *zdev,
 		data[i+2] = temp;
 	}
 
-	err = zocl_fpga_mgr_load(zdev->ddev, data, bit_header.BitstreamLength);
-	if (err)
-		DRM_ERROR("%s : ret code %d\n", __func__, err);
-
-	if (zdev->pr_isolation_addr && !IS_ERR_OR_NULL(map))
-		iowrite32(0x3, map);
-
-free_buffers:
-	if (map)
-		iounmap(map);
-	if (info)
-		fpga_image_info_free(info);
-	kfree(buffer);
-	kfree(bit_header.DesignName);
-	kfree(bit_header.PartName);
-	kfree(bit_header.Date);
-	kfree(bit_header.Time);
-	vfree(data);
-	return err;
-}
-
-int zocl_pcap_download_ioctl(struct drm_device *dev, void *data,
-			     struct drm_file *filp)
-{
-	struct axlf bin_obj;
-	char __user *buffer;
-	struct drm_zocl_dev *zdev = dev->dev_private;
-	const struct drm_zocl_pcap_download *args = data;
-	uint64_t primary_fw_off;
-	uint64_t primary_fw_len;
-	struct axlf_section_header *primaryHeader;
-
-	if (copy_from_user(&bin_obj, args->xclbin, sizeof(struct axlf)))
-		return -EFAULT;
-
-	if (memcmp(bin_obj.m_magic, "xclbin2", 8))
-		return -EINVAL;
-	
-	/* Check unique ID */
-	if (bin_obj.m_uniqueId == zdev->unique_id_last_bitstream) {
-		DRM_INFO("The XCLBIN already loaded. Don't need to reload.");
-		return 0;
+	/* Map PR address */
+	if (!zdev->pr_isolation_addr) {
+		DRM_ERROR("PR address is NULL");
+		return -ENODEV;
+	}
+	map = ioremap(zdev->pr_isolation_addr, 0x1000);
+	if (IS_ERR_OR_NULL(map)) {
+		DRM_ERROR("ioremap PR address failed");
+		return -ENOMEM;
 	}
 
-	primaryHeader = get_axlf_section(&bin_obj, BITSTREAM);
-	if (primaryHeader == NULL)
-		return -EINVAL;
-
-	if (zocl_check_section(primaryHeader, bin_obj.m_header.m_length,
-			       BITSTREAM))
-		return -EINVAL;
-
-	primary_fw_off = primaryHeader->m_sectionOffset;
-	primary_fw_len = primaryHeader->m_sectionSize;
-
-	buffer = (char __user *)args->xclbin;
-
-	if (!ZOCL_ACCESS_OK(VERIFY_READ, buffer, bin_obj.m_header.m_length))
-		return -EFAULT;
+	iowrite32(0x0, map);
```

Comment:
can we add some comments what we are trying to do here? Do we have risk here to write to some non exist address and hang the system?

Reply:
> can we add some comments what we are trying to do here? Do we have risk here to write to some non exist address and hang the system?

add comments